### PR TITLE
Fix unittest.TestCase assert methods' argument names

### PR DESCRIPTION
**Pull request by s-kovacevic**
Should solve #2716.

### CODE REVIEW

File: stdlib/3/unittest/__init__.pyi [hunk 1]
```diff
@@ -92,51 +88,56 @@ class TestCase:
...
                          msg: Any = ...) -> _AssertWarnsContext: ...
     def assertLogs(
         self, logger: Optional[logging.Logger] = ...,
         level: Union[int, str, None] = ...
     ) -> _AssertLogsContext: ...
     def assertAlmostEqual(self, first: float, second: float, places: int = ...,
                           msg: Any = ..., delta: float = ...) -> None: ...
+    @overload
+    def assertNotAlmostEqual(self, first: float, second: float,
+                             places: int = ..., msg: Any = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/unittest/__init__.pyi]: Why did you remove the `delta` argument? It seems to exist in recent Python 3.

Also, I'm not sure why the overloads are necessary.

File: stdlib/3/unittest/__init__.pyi [hunk 2]
```diff
@@ -92,51 +88,56 @@ class TestCase:
...
+                         expected_regex: Union[str, bytes, Pattern[str], Pattern[bytes]],
                          msg: Any = ...) -> _AssertWarnsContext: ...
     def assertLogs(
         self, logger: Optional[logging.Logger] = ...,
         level: Union[int, str, None] = ...
     ) -> _AssertLogsContext: ...
     def assertAlmostEqual(self, first: float, second: float, places: int = ...,
                           msg: Any = ..., delta: float = ...) -> None: ...
+    @overload
+    def assertNotAlmostEqual(self, first: float, second: float,
```
Comment (s-kovacevic) [inline stdlib/3/unittest/__init__.pyi]: Not sure about this one.

```python3
if delta is not None and places is not None:
    raise TypeError("specify delta or places not both")
```
https://github.com/python/cpython/blob/0f221d09cad46bee38d1b7a7822772df66c53028/Lib/unittest/case.py#L946-L947

Thats the reasoning, but not sure if typeshed should take into count things like this, let me know.

File: stdlib/3/unittest/__init__.pyi (hunk 1 above)
Comment (s-kovacevic) [reply stdlib/3/unittest/__init__.pyi]: Already commented.

File: stdlib/3/unittest/__init__.pyi (hunk 2 above)
Comment (JelleZijlstra) [reply stdlib/3/unittest/__init__.pyi]: Oh, I see. I think the current types aren't quite correct because delta is before msg on line 120 below, when in reality it should be after. It may be possible to get correct types if we use three overloads: one with neither place nor delta, one with only place, one with only delta. However, I'm also fine with just having a single entry in the stub.

File: stdlib/3/unittest/__init__.pyi [hunk 3]
```diff
@@ -92,51 +88,59 @@ class TestCase:
...
+                             msg: Any = ...) -> None: ...
```
[the remark this answers is not in the document]
Comment (s-kovacevic) [reply stdlib/3/unittest/__init__.pyi]: I see, same for `places` and `msg` in the second overload?

[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stdlib/3/unittest/__init__.pyi]: No, because those can be passed positionally.

[the remark this answers is not in the document]
Comment (s-kovacevic) [reply stdlib/3/unittest/__init__.pyi]: Oh yeah, my bad, it's getting late here. :smile:

**Comment by JelleZijlstra** [comment]
Thanks, looks good! I'll merge once CI is green.
